### ISSUE TITLE
Wrong Date Range behaviour in mobile resolution

### ISSUE THREAD

**Issue by KonradWroblewski** [Sage/carbon#5664]
### Current behaviour

With the minimum supported resolution of 320px, one element overlaps the other which is undesirable.

<img width="293" alt="image" src="https://user-images.githubusercontent.com/112560351/205021755-71bc04bc-2371-4545-a2b1-b5b983529217.png">

### Expected behaviour

<img width="306" alt="image" src="https://user-images.githubusercontent.com/112560351/205022312-8b317a7f-be38-4ebe-b146-d6364554d63a.png">

### CodeSandbox or Storybook URL

https://carbon.sage.com/?path=/docs/date-range--default-story

### JIRA Ticket (Sage Only)

SBS-35103

### Suggested Solution

I noticed that the margin between the elements is too large compared to the appearance in FIGMA (only in mobile resolution, in any other the margin is correct).

You can change this margin in mobile resolution or add props to modify this margin.

### Carbon Version

111.8.0

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Microsoft Edge

### What Operating System are you seeing the problem on?

Windows

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by Parsium** [comment]
Hi @KonradWroblewski, were you able to take a look at creating a small codesandbox example demonstrating this behaviour?

**Comment by Parsium** [comment]
Thanks for sharing this with us @KonradWroblewski

**Comment by Parsium** [comment]
Thanks for sharing this with us @KonradWroblewski 👍🏼 

Looking at the codesandbox it seems the container which wraps the two input boxes is wide enough at 320px to show the boxes side by side, but won't be if there is a certain amount of padding. Speaking with @ljemmo from Design System, it seems both the `DateInput` and `DateRange` components will be reviewed in the new year - to improve their accessibility and introduce design variants for low resolution (minimum 320px width) screens - so they suggest sticking with the vertically stacking behaviour for now as a temporary fix.

I must point out that we are also aware that calendar picker does not support low resolutions as it is too wide:
![Screenshot 2022-12-14 at 13 48 58](https://user-images.githubusercontent.com/18368713/207612017-bb8aaa99-3f2a-4b38-827c-0407ab6a3160.png)
We cannot currently address this until we have component designs to implement, but as mentioned above, Design System intend to work on this as part of their wider accessibility roadmap. 

I realise this raises a brand new issue for you, so may I ask if this work is urgent and this issue would block you?

**Comment by KonradWroblewski** [comment]
No. It doesn't stop us from working. So no problem, we can wait for the new version.

**Comment by Parsium** [comment]
> Thanks for sharing this with us @KonradWroblewski 👍🏼
> 
> Looking at the codesandbox it seems the container which wraps the two input boxes is wide enough at 320px to show the boxes side by side, but won't be if there is a certain amount of padding. Speaking with @ljemmo from Design System, it seems both the `DateInput` and `DateRange` components will be reviewed in the new year - to improve their accessibility and introduce design variants for low resolution (minimum 320px width) screens - so they suggest sticking with the vertically stacking behaviour for now as a temporary fix.
> 
> I must point out that we are also aware that calendar picker does not support low resolutions as it is too wide: ![Screenshot 2022-12-14 at 13 48 58](https://user-images.githubusercontent.com/18368713/207612017-bb8aaa99-3f2a-4b38-827c-0407ab6a3160.png) We cannot currently address this until we have component designs to implement, but as mentioned above, Design System intend to work on this as part of their wider accessibility roadmap in the new year.
> 
> I realise this raises a brand new issue for you, so may I ask if this will block any urgent work you may have?

Hi @clairedenning @ljemmo 👋🏼 Do we have a DSY ticket for this work mentioned here for the Date Picker and Date Range components in place?

**Comment by clairedenning** [comment]
@ingridzillinger Please can you reply to the comment above?

**Comment by ingridzillinger** [comment]
@Parsium  Added to Design System team's backlog: FT-139 (Works for Sage colleagues only):

**Comment by sianford** [comment]
Ticket DSY-2179 is still waiting to be assigned.

**Comment by ljemmo** [comment]
@sianford claire and i have 2 design tickets to pick up on general date picker enhancements such as accessibility improvements. We are lumping all of these improvements into these now. Here's mine: https://jira.sage.com/browse/DSY-2267

And here's claires: https://jira.sage.com/browse/DSY-2981

i'll close off DSY-2179

**Comment by edleeks87** [comment]
Closing this as new designs have been produced by DS and will be implemented in Carbon when the tickets are ready.
